src/Core/Services/DeviceTrustCryptoService.cs

```diff
@@ -77,5 +77,26 @@ private async Task<SymmetricCryptoKey> MakeDeviceKeyAsync()
             var randomBytes = await _cryptoFunctionService.RandomBytesAsync(DEVICE_KEY_SIZE);
             return new SymmetricCryptoKey(randomBytes);
         }
+
+        public async Task<bool> GetShouldTrustDeviceAsync()
+        {
+            return await _stateService.GetShouldTrustDeviceAsync();
+        }
+
+        public async Task SetShouldTrustDeviceAsync(bool value)
+        {
+            await _stateService.SetShouldTrustDeviceAsync(value);
+        }
+
+        public async Task<DeviceResponse> TrustDeviceIfNeededAsync()
+        {
+            DeviceResponse response = null;
+            if (await GetShouldTrustDeviceAsync())
+            {
+                response = await TrustDeviceAsync();
+                await SetShouldTrustDeviceAsync(false);
+            }
+            return response;
+        }
```

Comment:
⛏️ I don't see the `DeviceResponse` used anywhere and this can be written as follows:
```c#
public async Task TrustDeviceIfNeededAsync()
{
    if (!await GetShouldTrustDeviceAsync())
    {
        return null;
    }

    await TrustDeviceAsync();
    await SetShouldTrustDeviceAsync(false);
}
```
When it's something to be executed conditionally, I try to put the conditions at first to early return and most of the times it simplifies the code.

Reply:
yup you are totally right, my bad !